src/main/scala/tech/cryptonomic/conseil/tezos/TezosDatabaseOperations.scala

```diff
@@ -454,15 +455,20 @@ object TezosDatabaseOperations extends LazyLogging {
     predicates: List[Predicate],
     ordering: List[QueryOrdering],
     aggregation: List[Aggregation] = List.empty,
+    outputType: OutputType,
     limit: Int)
     (implicit ec: ExecutionContext): DBIO[List[QueryResponse]] = {
-     makeQuery(table, columns, aggregation)
+
+    val q = makeQuery(table, columns, aggregation)
        .addPredicates(aggregation.flatMap(_.getPredicate) ::: predicates)
        .addGroupBy(aggregation, columns)
        .addOrdering(ordering)
        .addLimit(limit)
-       .as[QueryResponse]
-       .map(_.toList)
+    if(outputType == OutputType.sql) {
+      DBIO.successful(List(Map("sql" -> Some(q.queryParts.mkString("")))))
+    } else {
+      q.as[QueryResponse].map(_.toList)
```

Comment:
As mentioned in the previous comment, I can't recall anymore why we have a List of Maps, can you help me refresh my memory?

Reply:
because one row returned from query is a `Map[ColumnName, Option(ColumnValue)]` and as we can have multiple rows returned we have `List[]` of them.

Reply:
oh, I see, we have many objects, each corresponding to a row... thanks